Meta/CMake/Superbuild/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.16)
+
+# Variables that we really don't want the user to pass to the serenity cross-build because they will screw things up :^)
+set(LAGOM_ONLY_VARS "CMAKE_CXX_COMPILER;CMAKE_C_COMPILER")
+set(EXCLUDE_FROM_CL_VARS "SERENITY_ARCH;SERENITY_TOOLCHAIN")
+
+# https://stackoverflow.com/a/48555098
+# https://cmake.org/pipermail/cmake/2018-January/067002.html
+# MUST be done before call to 'project'
+get_cmake_property(vars CACHE_VARIABLES)
+foreach(var IN LISTS vars)
+    get_property(currentHelpString CACHE "${var}" PROPERTY HELPSTRING)
+    if("${currentHelpString}" MATCHES "No help, variable specified on the command line." OR "${currentHelpString}" STREQUAL "")
+        message("${var} = [${${var}}]  --  ${currentHelpString}") # uncomment to see the variables being processed
+        if (var IN_LIST EXCLUDE_FROM_CL_VARS)
+            continue()
+        elseif(var IN_LIST LAGOM_ONLY_VARS)
+            list(APPEND LAGOM_CL_ARGS "-D${var}=${${var}}")
+        else()
+            list(APPEND CL_ARGS "-D${var}=${${var}}")
+        endif()
+    endif()
+endforeach()
+
+list(APPEND LAGOM_CL_ARGS ${CL_ARGS})
+
+project(
+    SerenitySuperbuild
+    DESCRIPTION "Orchestrate host and target builds in a single build"
+    LANGUAGES NONE
+)
+
+get_filename_component(
+    SERENITY_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../.."
+    ABSOLUTE CACHE
+)
+set(SERENITY_ARCH "i686" CACHE STRING "Architecture to build for (i686 or x86_64)")
+set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compliler toolchain to use for Serenity (GNU or Clang)")
+set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}")
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    lagom
+    SOURCE_DIR "${SERENITY_SOURCE_DIR}/Meta/Lagom"
+    BINARY_DIR "${PROJECT_BINARY_DIR}/../lagom"
+    INSTALL_DIR "${PROJECT_BINARY_DIR}/../lagom-install"
+    CMAKE_GENERATOR Ninja
```

Comment:
This is not necessary, the generator used to generate the superbuild is proxied by EP already.

Same below as well.

Reply:
I actually don't want to use the same generator. The build takes around five minutes with a cold ccache on my machine. If the super build uses `ninja` then invoking `cmake --build Build/superbuild` does not show any progress on each sub-build. That is, the build freezes on Building serenity.... For five minutes then dumps the build log for all 3800 files into the terminal. Instead of having the nice progress bar go across the screen. 

One solution I found for this was too use Makefiles for the superbuild, and keep using ninja for the sub-builds. If there's another one I'm all ears :). But I really don't want to have people come into discord like "Andrew I updated to your new build approach and it just freezes after installing the host tools" when it's actually doing the build, they're just not seeing it.

Reply:
> If the super build uses `ninja` then invoking `cmake --build Build/superbuild` does not show any progress on each sub-build.

That's weird, if you see the [fuzz step here](https://github.com/friendlyanon/cmake-init-gif-engine/runs/3450834777?check_suite_focus=true), you can see the output of both child builds, but also my [initial comment](https://github.com/SerenityOS/serenity/issues/4594#issuecomment-855310084) suggesting to move to a superbuild structure contained the output of Ninja builds that weren't spit out all at once.

A 5 minute build (assuming you have a beefy computer) doesn't sound terrible for my poor laptop, so I will investigate this one.

Reply:
https://user-images.githubusercontent.com/8388494/131226776-f50c6751-ac16-4072-b33e-722b5ca9e974.mp4

Here's the start of a superbuild build, note that it freezes until each stage is done, and then ninja dumps the entire output from that stage to the screen.

Reply:
https://user-images.githubusercontent.com/8388494/131227172-fb1251b9-72eb-4507-851e-ce356365c45c.mp4

Compared to an equivalent build with `Makefiles` as the generator. Gotta love `ccache` though :D.

Reply:
I found the cause and it seems to be a peculiarity of the Ninja generators only. https://github.com/friendlyanon/cmake-init-gif-engine/commit/f91c21ed3c1b74844cce534d4fd0ede0ba0225e8

`ccache` is disgusting indeed 👀